libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1446,6 +1446,28 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		_functionCall.annotation().type = make_shared<TupleType>(functionType->returnParameterTypes());
 
 	TypePointers parameterTypes = functionType->parameterTypes();
+
+	if (!_functionType.padArguments())
+	{
+		for (size_t i = 0; i < arguments.size(); ++i)
+		{
+			auto const& argType = type(*arguments[i]);
```

Comment:
this variable is not really required, you can just move that into the argument of the dynamic_cast.

Reply:
`argType` is also used in the message.

Reply:
Sure, but you can as well call the function on `literal` instead of on `argType`.